contracts/staking/contracts/src/sys/MixinParams.sol

```diff
@@ -86,29 +92,72 @@ contract MixinParams is
             uint256 _minimumPoolStake,
             uint256 _maximumMakersInPool,
             uint32 _cobbDouglasAlphaNumerator,
-            uint32 _cobbDouglasAlphaDenomintor
+            uint32 _cobbDouglasAlphaDenominator,
+            address _wethProxyAddress,
+            address _ethVaultAddress,
+            address _rewardVaultAddress,
+            address _zrxVaultAddress
         )
     {
         _epochDurationInSeconds = epochDurationInSeconds;
         _rewardDelegatedStakeWeight = rewardDelegatedStakeWeight;
         _minimumPoolStake = minimumPoolStake;
         _maximumMakersInPool = maximumMakersInPool;
         _cobbDouglasAlphaNumerator = cobbDouglasAlphaNumerator;
-        _cobbDouglasAlphaDenomintor = cobbDouglasAlphaDenomintor;
+        _cobbDouglasAlphaDenominator = cobbDouglasAlphaDenominator;
+        _wethProxyAddress = address(wethAssetProxy);
+        _ethVaultAddress = address(ethVault);
+        _rewardVaultAddress = address(rewardVault);
+        _zrxVaultAddress = address(zrxVault);
     }
 
-    /// @dev Assert param values before initializing them.
-    /// This must be updated for each migration.
-    function _assertMixinParamsBeforeInit()
+    /// @dev Initialzize storage belonging to this mixin.
+    /// @param _wethProxyAddress The address that can transfer WETH for fees.
+    /// @param _ethVaultAddress Address of the EthVault contract.
+    /// @param _rewardVaultAddress Address of the StakingPoolRewardVault contract.
+    /// @param _zrxVaultAddress Address of the ZrxVault contract.
+    function _initMixinParams(
+        address _wethProxyAddress,
+        address _ethVaultAddress,
+        address payable _rewardVaultAddress,
+        address _zrxVaultAddress
+    )
         internal
     {
         // Ensure state is uninitialized.
+        _assertParamsNotInitialized();
+
+        // Set up defaults.
+        // These cannot be set to variables, or we go over the stack variable limit.
+        _setParams(
+            2 weeks,                       // epochDurationInSeconds
```

Comment:
Nit: Rather than setting these to variables, it would probably be good to make these constants, in case we ever want to change them. Additionally, if they are constants, we can use the internal function pattern to stub out these values, if necessary.

Reply:
Had to do this to get around stack limitations unfortunately.